examples/structural_plasticity_with_stdp_2d.py

```diff
@@ -0,0 +1,267 @@
+"""
+Simple test for Structural Plasticity modified from a simple test for STDP.
+
+The current file integrates STDP and synaptic rewiring.
+
+
+To note:
+1. There is no initial connectivity
+2. Over time, at a rate of 10 kHz, rewiring is attempted, connectivity the
+    source and target layers
+3. When a connection is formed, it starts off with a weight of 0
+4. In the final testing phase enough connections have been formed to see
+    some post-synaptic spiking activity.
+
+author: Petrut Bogdan
+date  : March, 2018
+"""
+
+import numpy as np
+import pylab as plt
+import spynnaker8 as sim
+from pyNN.utility.plotting import Figure, Panel
+
+# SpiNNaker setup
+sim.setup(timestep=1.0, min_delay=1.0, max_delay=14)
+# +---------------------------------------------------------------------------+
+# | General Parameters                                                        |
+# +---------------------------------------------------------------------------+
+
+# Population parameters
+model = sim.IF_curr_exp
+sim.set_number_of_neurons_per_core(sim.IF_curr_exp, 100)
+cell_params = {'cm': 0.25,
+               'i_offset': 0.0,
+               'tau_m': 20.0,
+               'tau_refrac': 2.0,
+               'tau_syn_E': 5.0,
+               'tau_syn_I': 5.0,
+               'v_reset': -70.0,
+               'v_rest': -65.0,
+               'v_thresh': -50.0
+               }
+
+# Other simulation parameters
+e_rate = 80
+in_rate = 300
+
+n_stim_test = 5
+n_stim_pairing = 20
+dur_stim = 20
+
+# pop_size = 14**2
+pop_size = 7 ** 2
+
+ISI = 90.
+start_test_pre_pairing = 200.
+start_pairing = 1500.
+start_test_post_pairing = 700.
+
+simtime = (start_pairing + start_test_post_pairing
+           + ISI * (n_stim_pairing + n_stim_test) + 550.)
+
+# Initialisations of the different types of populations
+IAddPre = []
+IAddPost = []
+
+# +---------------------------------------------------------------------------+
+# | Creation of neuron populations                                            |
+# +---------------------------------------------------------------------------+
+
+# Neuron populations
+pre_pop = sim.Population(pop_size, model(**cell_params), label="PRE_SYN_POP")
+post_pop = sim.Population(pop_size, model(**cell_params), label="STRUCTURAL_POP")
+
+# Test of the effect of activity of the pre_pop population on the post_pop
+# population prior to the "pairing" protocol : only pre_pop is stimulated
+for i in range(n_stim_test):
+    IAddPre.append(sim.Population(pop_size,
```

Comment:
personally find this yucky, would rather have the params after pop on new line so that its not all down on right side of screen